test/suite/hook_nonstandalone.py

```diff
@@ -51,30 +51,18 @@ def get_platform_api(self):
         return self.platform_api
 
     # Is this test one we should skip?
-    def skip_test(self, test):
-        # Skip any test that contains one of these strings as a substring
-        skip = [
-                # Skip all tests that do timestamped truncate operations.
-                "test_checkpoint25.test_checkpoint",
-                "test_rollback_to_stable34.test_rollback_to_stable",
-                "test_rollback_to_stable36.test_rollback_to_stable36",
-                "test_truncate09.test_truncate09",
-                "test_truncate15.test_truncate15",
+    def skip_reason(self, test):
+        # There are no general categories of tests to skip. 
+        # Individual tests are skipped via the skip_for_hook decorator
+        return None
 
-                # This group fail within Python for various, sometimes unknown, reasons.
-                "test_bug018.test_bug018"
-                ]
-
-        for item in skip:
-            if item in str(test):
-                return True
-        return False
-
-    # Remove tests that won't work on non-standalone build
-    def filter_tests(self, tests):
-        new_tests = unittest.TestSuite()
-        new_tests.addTests([t for t in tests if not self.skip_test(t)])
-        return new_tests
+    # Skip tests that won't work on non-standalone build
+    def register_skipped_test(self, tests):
+        for t in tests:
+            skip_reason = self.skip_reason(t)
+            if skip_reason is not None:
```

Comment:
```suggestion
            if skip_reason:
```

Reply:
I originally did this as a type hint for what `skip_reason` returns, but decided on a refactor of the function.
The code has been updated in [5d029f7](https://github.com/wiredtiger/wiredtiger/pull/9764/commits/5d029f7d8b01a31619ba0f3448080c0b7564dc20) to return (bool, str) tuple instead